packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -4,31 +4,30 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+import { Org } from '@salesforce/core';
 import * as vscode from 'vscode';
 import { telemetryService } from '../telemetry';
 import { OrgAuthInfo, workspaceUtils } from '../util';
+import { WorkspaceContext } from './workspaceContext';
 
 export enum OrgType {
   SourceTracked,
   NonSourceTracked
 }
 
-export async function getWorkspaceOrgType(
-  defaultUsernameOrAlias?: string
-): Promise<OrgType> {
-  if (!defaultUsernameOrAlias) {
-    const e = new Error();
-    e.name = 'NoDefaultusernameSet';
-    throw e;
-  }
-  const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias);
-  const isScratchOrg = await OrgAuthInfo.isAScratchOrg(username).catch(err =>
-    telemetryService.sendException(
-      'get_workspace_org_type_scratch_org',
-      err.message
-    )
-  );
-  return isScratchOrg ? OrgType.SourceTracked : OrgType.NonSourceTracked;
+/**
+ * @description determines whether the default org is source-tracked or not.
+ * During dev it was observed that there were some potential issues with other options
+ * (org.isScratch, org.tracksSource) related to cache-ing and a newly created
+ * Scratch Org would sometimes return false.  Using org.supportsSourceTracking()
+ * because it has been the most consistently accurate solution here.
+ * @returns OrgType (SourceTracked or NonSourceTracked) of the current default org
+ */
+export async function getWorkspaceOrgType(): Promise<OrgType> {
+  const connection = await WorkspaceContext.getInstance().getConnection();
+  const org: Org = await Org.create({ connection });
```

Comment:
On line 28, the type definition comes from `Org.create()`, and the `: Org` isn't needed.  preferred change would be:

`const org = await Org.create({ connection });`